### PR TITLE
Create Detection for Password Spray against ADFSSignInLogs

### PR DESCRIPTION
**Pull request by ep3p**
Change(s):
   - A detection for ADFSSignInLogs.

   Version Updated:
   - Does not apply

   Testing Completed:
   - Yes

   Checked that the validations are passing and have addressed any issues that are present:
   - ADFSSignInLogs is not a correct dataType yet.

-----------------------------------------------------------------------------------------------------------
Feel free to change query frequency/period.

Please, could you check other scenarios where this detection might be a false positive? (for example depending on Authentication Details)

### CODE REVIEW

**Comment by ep3p** [comment]
I have observed that ADFSSignInLogs might not arrive instantaneously, and thus possible alerts might be lost. So the query should wait a bit for the logs to arrive, and maybe the queryfrequency should be increased.

**Comment by ep3p** [comment]
Substituted the query_wait by ingestion_time()

**Comment by vmanojreddy** [comment]
@aprakash13 Please review Detections. Thanks.

File: Detections/SigninLogs/ADFSSignInLogsPasswordSpray.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+queryFrequency: 30m
+queryPeriod: 1h
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - CredentialAccess
+relevantTechniques:
+  - T1110
+query: |
+
```
Comment (aprakash13) [inline Detections/SigninLogs/ADFSSignInLogsPasswordSpray.yaml:20]: Please remove the additional line. Not needed.

File: Detections/SigninLogs/ADFSSignInLogsPasswordSpray.yaml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+      DistinctFailureCount = dcountif(UserPrincipalName, ResultType !in (successCodes)),
+      DistinctSuccessCount = dcountif(UserPrincipalName, ResultType in (successCodes)),
+      SuccessAccounts = make_set_if(UserPrincipalName, ResultType in (successCodes), 250),
+      arg_min(TimeGenerated, Category, OperationName, AuthenticationRequirement, ConditionalAccessStatus, IsInteractive, UserAgent, NetworkLocationDetails, DeviceDetail, TokenIssuerType, TokenIssuerName, ResourceIdentity)
+      by IPAddress
+  | where DistinctFailureCount > DistinctSuccessCount and DistinctFailureCount >= accountthreshold
+  //| extend SuccessAccounts = iff(array_length(SuccessAccounts) != 0, SuccessAccounts, dynamic(["null"]))
+  //| mv-expand SuccessAccounts
+  | project TimeGenerated, Category, OperationName, IPAddress, DistinctFailureCount, DistinctSuccessCount, SuccessAccounts, AuthenticationRequirement, ConditionalAccessStatus, IsInteractive, UserAgent, NetworkLocationDetails, DeviceDetail, TokenIssuerType, TokenIssuerName, ResourceIdentity
+
```
Comment (aprakash13) [inline Detections/SigninLogs/ADFSSignInLogsPasswordSpray.yaml:38]: Please remove the additional line. Not needed.

**Review by aprakash13 — CHANGES_REQUESTED** [review]
Thanks for all your contributions @ep3p. Can you please fix the validation issues before we can merge these? Also, a couple of very minor things to look at.

**Comment by ep3p** [comment]
> Thanks for all your contributions @ep3p. Can you please fix the validation issues before we can merge these? Also, a couple of very minor things to look at.

@aprakash13 sorry I don't know yet how to fix this, ADFSSignInLogs is not being recognized as a valid table by KQL Validations yet. I guess somewhere in this repo "ADFSSignInLogs" string could be added a valid parameter for Analytic Rules templates, but I don't know where exactly, could you guide me if you want me to mess with that?

https://dev.azure.com/azure/Azure-Sentinel/_build/results?buildId=40174&view=logs&j=2bbe2297-1b5b-54f3-6060-67e7b289f6c3&t=293f4a23-e36d-5efa-3a4e-ea46d4ab6d83&l=27

**Comment by ep3p** [comment]
@aprakash13 I have opened a PR for the custom table for KQL validation for ADFSSignInLogs

https://github.com/Azure/Azure-Sentinel/pull/4700

**Comment by aprakash13** [comment]
> @aprakash13 I have opened a PR for the custom table for KQL validation for ADFSSignInLogs
> 
> #4700

Thanks @ep3p . Once the Custom tables in https://github.com/Azure/Azure-Sentinel/pull/4700 gets merged. We can move forward with this.

**Review by aprakash13 — APPROVED** [review]
Thanks @ep3p for all your contributions.

**Pull request merged**